[libtirpc] [host] fails to build

Build failure of this package using the latest HEAD.

Complete build log:  https://gist.github.com/graysky2/d1b904d824208c05ac75e50a1f843c5f

Excerpt of error:
```
% make package/libtirpc/host/compile V=s
...
auth_none.c:65:15: error: conflicting types for 'xdr_opaque_auth'; have 'bool_t(void)' {aka 'int(void)'}
   65 | extern bool_t xdr_opaque_auth();
      |               ^~~~~~~~~~~~~~~
In file included from auth_none.c:51:
../tirpc/rpc/auth.h:303:15: note: previous declaration of 'xdr_opaque_auth' with type 'bool_t(XDR *, struct opaque_auth *)' {aka 'int(struct __rpc_xdr *, struct opaque_auth *)'}
  303 | extern bool_t xdr_opaque_auth           (XDR *, struct opaque_auth *);
      |               ^~~~~~~~~~~~~~~
auth_none.c: In function 'authnone_create':
auth_none.c:98:23: error: too many arguments to function 'xdr_opaque_auth'; expected 0, have 2
   98 |                 (void)xdr_opaque_auth(xdrs, &ap->no_client.ah_cred);
      |                       ^~~~~~~~~~~~~~~ ~~~~
auth_none.c:65:15: note: declared here
   65 | extern bool_t xdr_opaque_auth();
      |               ^~~~~~~~~~~~~~~
auth_none.c:99:23: error: too many arguments to function 'xdr_opaque_auth'; expected 0, have 2
   99 |                 (void)xdr_opaque_auth(xdrs, &ap->no_client.ah_verf);
      |                       ^~~~~~~~~~~~~~~ ~~~~
auth_none.c:65:15: note: declared here
   65 | extern bool_t xdr_opaque_auth();
      |               ^~~~~~~~~~~~~~~
svc_auth_none.c:44:9: error: initialization of 'int (*)(struct SVCAUTH *, XDR *, bool_t (*)(XDR *, ...), char *)' {aka 'int (*)(struct SVCAUTH *, struct __rpc_xdr *, int (*)(struct __rpc_xdr *, ...), char *)'} from incompatible pointer type 'bool_t (*)(void)' {aka 'int (*)(void)'} [-Wincompatible-pointer-types]
   44 |         svcauth_none_wrap,
      |         ^~~~~~~~~~~~~~~~~
svc_auth_none.c:44:9: note: (near initialization for 'svc_auth_none_ops.svc_ah_wrap')
svc_auth_none.c:41:17: note: 'svcauth_none_wrap' declared here
   41 | static bool_t   svcauth_none_wrap();
      |                 ^~~~~~~~~~~~~~~~~
svc_auth_none.c:45:9: error: initialization of 'int (*)(struct SVCAUTH *, XDR *, bool_t (*)(XDR *, ...), char *)' {aka 'int (*)(struct SVCAUTH *, struct __rpc_xdr *, int (*)(struct __rpc_xdr *, ...), char *)'} from incompatible pointer type 'bool_t (*)(void)' {aka 'int (*)(void)'} [-Wincompatible-pointer-types]
   45 |         svcauth_none_wrap,
      |         ^~~~~~~~~~~~~~~~~
svc_auth_none.c:45:9: note: (near initialization for 'svc_auth_none_ops.svc_ah_unwrap')
svc_auth_none.c:41:17: note: 'svcauth_none_wrap' declared here
   41 | static bool_t   svcauth_none_wrap();
      |                 ^~~~~~~~~~~~~~~~~
svc_auth_none.c:46:9: error: initialization of 'int (*)(struct SVCAUTH *)' from incompatible pointer type 'bool_t (*)(void)' {aka 'int (*)(void)'} [-Wincompatible-pointer-types]
   46 |         svcauth_none_destroy
      |         ^~~~~~~~~~~~~~~~~~~~
svc_auth_none.c:46:9: note: (near initialization for 'svc_auth_none_ops.svc_ah_destroy')
svc_auth_none.c:40:17: note: 'svcauth_none_destroy' declared here
   40 | static bool_t   svcauth_none_destroy();
      |                 ^~~~~~~~~~~~~~~~~~~~
svc_auth_none.c:55:1: error: conflicting types for 'svcauth_none_destroy'; have 'bool_t(SVCAUTH *)' {aka 'int(SVCAUTH *)'}
   55 | svcauth_none_destroy(SVCAUTH *auth)
      | ^~~~~~~~~~~~~~~~~~~~
svc_auth_none.c:40:17: note: previous declaration of 'svcauth_none_destroy' with type 'bool_t(void)' {aka 'int(void)'}
   40 | static bool_t   svcauth_none_destroy();
      |                 ^~~~~~~~~~~~~~~~~~~~
svc_auth_none.c:61:1: error: conflicting types for 'svcauth_none_wrap'; have 'bool_t(SVCAUTH *, XDR *, bool_t (*)(XDR *, ...), char *)' {aka 'int(SVCAUTH *, struct __rpc_xdr *, int (*)(struct __rpc_xdr *, ...), char *)'}
   61 | svcauth_none_wrap(SVCAUTH *auth, XDR *xdrs, xdrproc_t xdr_func,
      | ^~~~~~~~~~~~~~~~~
svc_auth_none.c:41:17: note: previous declaration of 'svcauth_none_wrap' with type 'bool_t(void)' {aka 'int(void)'}
   41 | static bool_t   svcauth_none_wrap();
      |                 ^~~~~~~~~~~~~~~~~
svc_auth_none.c:40:17: warning: 'svcauth_none_destroy' used but never defined
   40 | static bool_t   svcauth_none_destroy();
      |                 ^~~~~~~~~~~~~~~~~~~~
svc_auth_none.c:41:17: warning: 'svcauth_none_wrap' used but never defined
   41 | static bool_t   svcauth_none_wrap();
      |                 ^~~~~~~~~~~~~~~~~
svc_auth_none.c:61:1: warning: 'svcauth_none_wrap' defined but not used [-Wunused-function]
   61 | svcauth_none_wrap(SVCAUTH *auth, XDR *xdrs, xdrproc_t xdr_func,
      | ^~~~~~~~~~~~~~~~~
svc_auth_none.c:55:1: warning: 'svcauth_none_destroy' defined but not used [-Wunused-function]
   55 | svcauth_none_destroy(SVCAUTH *auth)
      | ^~~~~~~~~~~~~~~~~~~~
auth_time.c: In function 'alarm_hndler':
auth_time.c:61:1: warning: old-style function definition [-Wold-style-definition]
   61 | alarm_hndler(s)
      | ^~~~~~~~~~~~
auth_time.c: In function 'uaddr_to_sockaddr':
auth_time.c:83:12: warning: old-style function definition [-Wold-style-definition]
   83 | static int uaddr_to_sockaddr(uaddr, sin)
      |            ^~~~~~~~~~~~~~~~~
auth_time.c: In function 'free_eps':
auth_time.c:118:1: warning: old-style function definition [-Wold-style-definition]
  118 | free_eps(eps, num)
      | ^~~~~~~~
auth_time.c: In function 'get_server':
auth_time.c:144:1: warning: old-style function definition [-Wold-style-definition]
  144 | get_server(sin, host, srv, eps, maxep)
      | ^~~~~~~~~~
auth_time.c: In function '__rpc_get_time_offset':
auth_time.c:229:1: warning: old-style function definition [-Wold-style-definition]
  229 | __rpc_get_time_offset(td, srv, thost, uaddr, netid)
      | ^~~~~~~~~~~~~~~~~~~~~
getpublickey.c: In function '__getpublickey_real':
getpublickey.c:61:1: warning: old-style function definition [-Wold-style-definition]
   61 | __getpublickey_real(netname, publickey)
      | ^~~~~~~~~~~~~~~~~~~
getpublickey.c: In function 'getpublicandprivatekey':
getpublickey.c:88:1: warning: old-style function definition [-Wold-style-definition]
   88 | getpublicandprivatekey(key, ret)
      | ^~~~~~~~~~~~~~~~~~~~~~
getpublickey.c: In function 'getpublickey':
getpublickey.c:162:5: warning: old-style function definition [-Wold-style-definition]
  162 | int getpublickey(netname, publickey)
      |     ^~~~~~~~~~~~
getpublickey.c:167:24: error: too many arguments to function '__getpublickey_LOCAL'; expected 0, have 2
  167 |                 return(__getpublickey_LOCAL(netname, publickey));
      |                        ^~~~~~~~~~~~~~~~~~~~ ~~~~~~~
auth_time.c:470:33: error: passing argument 2 of 'signal' from incompatible pointer type [-Wincompatible-pointer-types]
  470 |                 signal(SIGALRM, oldsig);
      |                                 ^~~~~~
      |                                 |
      |                                 void (*)(void)
In file included from auth_time.c:37:
/usr/include/signal.h:88:57: note: expected '__sighandler_t' {aka 'void (*)(int)'} but argument is of type 'void (*)(void)'
   88 | extern __sighandler_t signal (int __sig, __sighandler_t __handler)
      |                                          ~~~~~~~~~~~~~~~^~~~~~~~~
/usr/include/signal.h:72:16: note: '__sighandler_t' declared here
   72 | typedef void (*__sighandler_t) (int);
      |                ^~~~~~~~~~~~~~
make[5]: *** [Makefile:923: libtirpc_la-svc_auth_none.lo] Error 1
```

--- a/libs/libtirpc/Makefile
+++ b/libs/libtirpc/Makefile
@@ -2,7 +2,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=libtirpc
 PKG_VERSION:=1.3.4
-PKG_RELEASE:=1
+PKG_RELEASE:=2
 
 PKG_SOURCE_URL:=@SF/libtirpc
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
@@ -35,7 +35,7 @@ HOST_CONFIGURE_ARGS += --disable-symvers
 endif
 
 TARGET_CFLAGS += -DGQ
-HOST_CFLAGS += -DGQ
+HOST_CFLAGS += -DGQ -std=c99
 
 define Package/libtirpc/install
 	$(INSTALL_DIR) $(1)/usr/lib

